### PR TITLE
Support more branch name formats

### PR DESCRIPTION
**Pull request by paour**
Allow setting a Jira project so the branch name detection is more lenient.

Addresses #6 (except the part about passing in the issue ID.

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -42,6 +42,14 @@ inputs:
       is_expand: true
       is_required: true
       value_options: []
+  - jira_project:
```
Comment (dagio) [inline step.yml:45]: Is it the JIRA project key ? If so maybe we should make it explicit in the summary / description ?

File: tests/IssueKeyResolverTest.php [hunk 2]
```diff
@@ -12,6 +12,7 @@ public function dataProvider()
         return [
             ['FOO-123', 'feature/FOO-123-Add-Something'],
             ['FOO-123', 'hotfix/FOO-123-Add-Something'],
+            ['FOO-123', 'bugfix/FOO-123-Add-Something'],
```
Comment (dagio) [inline tests/IssueKeyResolverTest.php:15]: The following case is missing : `['FOO-123', 'bug/FOO-123-Add-Something'],`

File: tests/IssueKeyResolverTest.php [hunk 3]
```diff
@@ -34,6 +35,19 @@ public function testIssueKeyResolved($expectedIssueKey, $branchName)
     }
 
     /**
+     * @param string $issueKey
+     * @param string $branchName
+     *
+     * @dataProvider dataProvider
+     */
+    public function testIssueKeyResolvedWithProject($expectedIssueKey, $branchName)
```
Comment (dagio) [inline tests/IssueKeyResolverTest.php:44]: In that test you alter the branch name that is tested. It's not clear what you are trying to achieve. Maybe the following would make it more explicit :

```
$resolver = new IssueKeyResolver();
$branchNameWithoutProjectKey = str_replace('FOO-', '', $branchName);
$projectKey = 'FOO';
$actualIssueKey = $resolver->resolveKeyFromBranchName($branchNameWithoutProjectKey, $projectKey);
$this->assertEquals($expectedIssueKey, $actualIssueKey);
```

**Review by dagio — CHANGES_REQUESTED** [review]
Thanks for the PR. Just few feedbacks

**Comment by paour** [comment]
Done, thank you!

I didn't want to change composer files in a PR, but the repo seems to have some composer files (`vendor`, composer.lock`) that I imagine should be removed and added to `.gitignore` (but I know nothing about Composer…).

**Review by dagio — APPROVED** [review]

**Comment by dagio** [comment]
I kept them in order to avoid to have to run `composer install` an slow down the build. We should actually just use a phar. That could be an improvement

**Pull request merged**